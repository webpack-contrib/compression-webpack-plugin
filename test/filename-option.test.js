import Plugin from '../src/index';

import {
  compile,
  getAssetsNameAndSize,
  getCompiler,
  getErrors,
  getWarnings,
} from './helpers/index';

describe('"filename" option', () => {
  let compiler;

  beforeEach(() => {
    compiler = getCompiler(
      './entry.js',
      {},
      {
        output: {
          path: `${__dirname}/dist`,
          filename: '[name].js?var=[hash]',
          chunkFilename: '[id].[name].js?ver=[hash]',
        },
      }
    );
  });

  it('matches snapshot for `[path].super-compressed.gz[query]` value ({String})', async () => {
    new Plugin({
      minRatio: 1,
      filename: '[path].super-compressed.gz[query]',
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });

  it('matches snapshot for `[dir][name].super-compressed.gz[ext][query]` value ({String})', async () => {
    new Plugin({
      minRatio: 1,
      filename: '[dir][name].super-compressed.gz[ext][query]',
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });

  it('matches snapshot for custom function ({Function})', async () => {
    new Plugin({
      minRatio: 1,
      filename(info) {
        return `${info.path}.gz${info.query}`;
      },
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(getAssetsNameAndSize(stats)).toMatchSnapshot('assets');
    expect(getWarnings(stats)).toMatchSnapshot('errors');
    expect(getErrors(stats)).toMatchSnapshot('warnings');
  });
});
