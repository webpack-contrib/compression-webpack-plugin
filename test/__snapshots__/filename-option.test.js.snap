// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`function\` option matches snapshot for \`[path].super-compressed.gz[query]\` value ({String}): assets 1`] = `
Array [
  Array [
    "0.async.js.super-compressed.gz?ver=81982f432266ef1edd1a",
    117,
  ],
  Array [
    "0.async.js?ver=81982f432266ef1edd1a",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.super-compressed.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.super-compressed.gz",
    393,
  ],
  Array [
    "js.js.super-compressed.gz?var=81982f432266ef1edd1a",
    2802,
  ],
  Array [
    "js.js?var=81982f432266ef1edd1a",
    10221,
  ],
]
`;

exports[`when applied with \`function\` option matches snapshot for \`[path].super-compressed.gz[query]\` value ({String}): errors 1`] = `Array []`;

exports[`when applied with \`function\` option matches snapshot for \`[path].super-compressed.gz[query]\` value ({String}): warnings 1`] = `Array []`;

exports[`when applied with \`function\` option matches snapshot for custom function ({Function}): assets 1`] = `
Array [
  Array [
    "0.async.js.gz?ver=81982f432266ef1edd1a",
    117,
  ],
  Array [
    "0.async.js?ver=81982f432266ef1edd1a",
    130,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png.gz",
    73160,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    393,
  ],
  Array [
    "js.js.gz?var=81982f432266ef1edd1a",
    2802,
  ],
  Array [
    "js.js?var=81982f432266ef1edd1a",
    10221,
  ],
]
`;

exports[`when applied with \`function\` option matches snapshot for custom function ({Function}): errors 1`] = `Array []`;

exports[`when applied with \`function\` option matches snapshot for custom function ({Function}): warnings 1`] = `Array []`;
